pkgs/build-support/setup-hooks/separate-debug-info.sh

```diff
@@ -43,10 +43,11 @@ _separateDebugInfo() {
             then
                 echo "separate-debug-info: warning: multiple files with build id $id found, overwriting"
             fi
-            $OBJCOPY --only-keep-debug "$i" "$dst/${id:0:2}/${id:2}.debug"
-
-            # Also a create a symlink <original-name>.debug.
-            ln -sfn ".build-id/${id:0:2}/${id:2}.debug" "$dst/../$(basename "$i")"
+            (
+                $OBJCOPY --only-keep-debug "$i" "$dst/${id:0:2}/${id:2}.debug" &&
+                # Also a create a symlink <original-name>.debug.
+                ln -sfn ".build-id/${id:0:2}/${id:2}.debug" "$dst/../$(basename "$i")"
```

Comment:
I don't understand why we need to do this.  Isn't stdenv -e by default?

Reply:
I assume not? Because this does not hard fail the build, but it does fail to create the debug info file, which results in a dangling symlink.

Reply:
But like half of Nixpkgs requires on build failing if a shell command does

Reply:
https://github.com/NixOS/nixpkgs/blob/25f794db5887e7edf40afc3b1f0593d3d7de4ef8/pkgs/build-support/setup-hooks/separate-debug-info.sh#L51

oops.

Reply:
So it's not -e inside the subshell?

Reply:
It's not, subshells don't inherit these.

Reply:
Ugh, TIL. Might be a little less confusing to set -e inside the subshell, but probably not important.